### PR TITLE
Expands Details view, adding parcels, features, and investments

### PR DESCRIPTION
**Pull request by allthesignals**
#18 and #19 should be merged before this one is merged.

This PR further expands on #12. It adds many new routes, controllers, models, and complementary mirage models, factories, and configuration actions. 

Here's a breakdown of what's going on:
Map state is handled in a top-level controller and template:
templates/cities.hbs
controllers/cities.js

These routes manage cities index and show route state, as well as associations:
routes/cities.js
routes/cities/city.js
routes/cities/city/details.js

The details route grabs the associations for that city only and sets them on the top-level controller for the map. The `willTransition()` hook clears the top-level cities controllers. This is for performance reasons - showing all 600 points with al the Ember bindings and things going on was far too slow. (As a side note, I'm thinking about the Carto stuff as an enhancement - this is a good compromise for now, unless @cball thinks otherwise).

Some remaining issues: the app creates n+1 new requests for the associations. I'd rather side load (`cities/1?includes=features`) or do whatever this is called: `cities/1/features`.

### CODE REVIEW

File: app/controllers/cities/city/details.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import Ember from 'ember';
+
+export default Ember.Controller.extend({
+  citiesController: Ember.inject.controller('cities'),
+  currentFeatures: Ember.computed.alias('citiesController.currentFeatures'),
+  currentInvestments: Ember.computed.alias('citiesController.currentFeatures')
```
Comment (cball) [inline app/controllers/cities/city/details.js]: We should talk through this one when you get time. I think the service comment in the previous PR would help here so you're not having to inject a controller.

File: app/models/investment.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+export default DS.Model.extend({
+  name: DS.attr("string"),
+  address: DS.attr("string"),
+  contact: DS.attr("string"),
+  employer: DS.attr("string"),
+  activating: DS.attr("string"),
+  city: DS.belongsTo("city"),
+  latitude: DS.attr("number"),
+  longitude: DS.attr("number")
+});
```
Comment (cball) [inline app/models/investment.js]: for organization purposes, you might want to keep the attributes at the top and relationships at the bottom.

File: app/routes/cities/city/details.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    controller.set('currentInvestments', city.get('investments'));
+    controller.set('currentParcels', city.get('parcels'));
+  },
+  actions: {
+    willTransition() {
+      let controller = this.controllerFor('cities');
+      controller.set('currentFeatures', null);
+      controller.set('currentInvestments', null);
+      controller.set('currentParcels', null);
+    }
```
Comment (cball) [inline app/routes/cities/city/details.js]: This definitely points to a service that houses all of the current selections.

File: mirage/factories/city.js [hunk 4]
```diff
@@ -1,7 +1,25 @@
...
+  min = Math.ceil(min);
+  max = Math.floor(max);
+  return Math.floor(Math.random() * (max - min)) + min;
+}
 
 export default Factory.extend({
   name() { return faker.address.city(); },
   fellows() { return faker.random.boolean(); },
-  avatar() { return faker.internet.avatar(); }
+  avatar() { return faker.internet.avatar(); },
```
Comment (cball) [inline mirage/factories/city.js]: all of these can simplify to:

```js
avatar: faker.internet.avatar
```

Mirage will call the function when it creates the factory.

**Comment by allthesignals** [comment]
@cball I made an issue about services here: #29. As you can see from [wmg/23-create](/allthesignals/cityxcity/tree/wmg/23-create), there's lots of inelegance that stems from the issue pointed out in https://github.com/allthesignals/cityxcity/pull/27#discussion_r94268128. Let's definitely talk about a plan going forward, because I know during the create/update/delete part of Ember apps, I get into some trouble with controllers, routes, and things. 

On top of that, there are 3 associations on `city`: `features`, `parcels`, `investments`, which need to be CRUDable. Should be easy, except there would be a lot of duplication in respective controllers (due to the need for services) and I'm not sure how the route structure should work just yet. 

In summary, main issues:

1. Refactor "only display current city associations" logic (as mentioned in https://github.com/allthesignals/cityxcity/pull/27#discussion_r94268128).
2. Talk about plan for route structure and how (1) affects it.
3. Plan out CRUD for `city` model associations `features`, `parcels`, and `investments`.

Thanks! And have a good new year.

File: mirage/factories/city.js (hunk 4 above)
Comment (allthesignals) [inline mirage/factories/city.js]: Will do! Faker seems to struggle with that sometimes though: https://github.com/samselikoff/ember-cli-mirage/issues/306. Don't know if you've encountered that.

File: app/routes/cities/city/details.js (hunk 3 above)
Comment (allthesignals) [inline app/routes/cities/city/details.js]: I went ahead and tried the service object, and it seems to be working great. See updated code.
